src/rust/src/parport/chardev.rs

```diff
@@ -43,7 +54,7 @@ impl NkCharDev {
         }
     }
 
-    pub fn register(&mut self, parport: Arc<IRQLock<Parport>>) -> Result {
+    pub fn register(&mut self, deivce: Arc<IRQLock<T>>) -> Result<()> {
```

Comment:
Let's use `Result` instead of `Result<()>` to be consistent with the rest of the code base.